### DISCUSSION TITLE
Custom migration path for testing database tests?

### DISCUSSION THREAD

**Discussion by shirkerohit** [laravel/framework#38388] [Q&A]
Hey,

I am in a scenario where the migrations are scattered across different folders as per modules and the test that I am running with refreshDatabase only refreshes the database with migration from the default folder.

I know we can have an artisan call in the test with the migration path but then it somehow overrides the purpose of having RefreshDatabase Trait. What about having a default method/property on that check if any custom path is defined and If there is any use it else falls back to default?

Here is something I am thinking? worth PR?

```
/**
     * The parameters that should be used when running "migrate:fresh".
     *
     * @return array
     */
    protected function migrateFreshUsing()
    {
        $seeder = $this->seeder();

        return array_merge(
            [
                '--drop-views' => $this->shouldDropViews(),
                '--drop-types' => $this->shouldDropTypes(),
                '--path' => $this->useMigrationPath(),
            ],
            $seeder ? ['--seeder' => $seeder] : ['--seed' => $this->shouldSeed()],
        );
    }


/**
     * return a custom migration path else fall back to the default path.
     *
     * @return string
     */
    protected function useMigrationPath()
    {
        if (method_exists($this, 'useMigrationPath')) {
            return $this->useMigrationPath();
        }
        return database_path();
    }

```

Does it make any sense? Please suggest...

**Comment by shirkerohit** [comment]
Closing this since there is no reply!

**Comment by shaedrich** [comment]
I actually had that case because I wanted to keep the migrations folder small. So, I'd like that.
